test/e2e/workflow/deployment_test.go

```diff
@@ -281,25 +281,57 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			CreateConfig(gvk, configSpec)
 			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, 30*time.Second)
 		})
-		Context("and checking CNAO prometheus endpoint", func() {
-			var err error
-			var scrapedData string
-			BeforeEach(func() {
-				scrapedData, err = GetScrapedDataFromMonitoringEndpoint()
+	})
+	Context("when deploying components and checking CNAO prometheus endpoint", func() {
+		type prometheusScrapeParams struct {
+			configSpec             cnao.NetworkAddonsConfigSpec
+			expectedMetricValueMap map[string]string
+		}
+		DescribeTable("and checking scraped data",
+			func(p prometheusScrapeParams) {
```

Comment:
can the func get 2 parameters instead of the struct ?
i don't see a reason to encapsulate those parameters into a struct

Reply:
I prefer it like this. 
helps keep parameter names understood , and makes thing neater imo.